### ISSUE TITLE
vhost config getting ignored using an fqdn

### ISSUE THREAD

**Issue by c0debreaker** [expressjs/vhost#25]
I have this config. When I load our app, we make a call to `ws://iptv.exampe.org/ws?iptv_config,iptv_channelStatus`. However, the vhost config doesn't intercept it. It is still sending the request to my `hpmToLocalProxy` which is not our Socksv5 proxy server. I need to forward all request to SocksV5 server if domain is iptv.example.org including websockets which is what I'm currently debugging. What am I doing wrong?

```
SocksProxy = {
  agent: new Socks5Agent({
    socksHost: 'socksv5proxy.example.org',
    socksPort: 9050
  }),
  target: 'iptv.example.org',
  changeOrigin: true,
  ws: true,
  logLevel: 'debug'
};
vacpuSocks5ProxyConfig = proxy(SocksProxy);
app.use(vhost('iptv.example.org', Socks5ProxyConfig));

  hpmToLocalProxy = {
    target: 'http://www.example.org',
    changeOrigin: true,
    router: proxyHelper.customNonProxyRouter,
    secure: false,
    ws: true,
    logLevel: 'debug'
  };

hpmToLocalProxyConfig = proxy(hpmToLocalProxy);
app.use('/path1', hpmToLocalProxyConfig);
app.use('/path2, hpmToLocalProxyConfig);
app.use('/, hpmToLocalProxyConfig);

```

**Comment by dougwilson** [comment]
Hi @c0debreaker hm, I'm not sure what would be wrong. I tried to get your code running, but there is a lot of references to things I don't know how to reproduce on my end. Can you help provide the following so I can get it running and try to see what is going on?

1. Version of Node.js
2. Version of this module and all other modules needed to run your code
3. Complete code I can copy and paste
4. Complete instructions for how to get the code running
5. Instructions on how to reproduce the issue

Thanks!

**Comment by c0debreaker** [comment]
It's ok now. I figured out a way by creating an aliased ip on loopback pointing to 127.0.0.2 then I assigned my server listener with that ip. This way, I don't have to vhost anymore.

in /etc/hosts
```
127.0.0.1 mainapp.example.com
127.0.0.2 api.example.com
```

When the request is api.example.com, the appropriate tcp listener listens. http-proxy-middleware is now finally not getting confused. Previously, http-proxy-middleware was forwarding to mainapp.example.com even if api.example.com was the call.
